### PR TITLE
[REEF-1320] Make default group creation passive

### PR DESCRIPTION
**Pull request by jwang98052**
JIRA: [REEF-1320](https://issues.apache.org/jira/browse/REEF-1320)
This closes #

### CODE REVIEW

**Comment by jwang98052** [comment]
@dkm2110 @markusweimer This is only for default group. Can you have a quick look?

File: lang/cs/Org.Apache.REEF.Network/Group/Driver/Impl/GroupCommDriver.cs [hunk 1]
```diff
@@ -101,7 +102,20 @@ public sealed class GroupCommDriver : IGroupCommDriver
...
         {
-            get { return _commGroups[_groupName]; }
+            get
+            {
+                lock (_groupsLock)
+                {
+                    ICommunicationGroupDriver defaultGroup;
+                    _commGroups.TryGetValue(_defaultGroupName, out defaultGroup);
+
+                    if (defaultGroup == null)
```
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Network/Group/Driver/Impl/GroupCommDriver.cs]: Why not the following: 

```
if (_defaultGroupName == null)
{
  NewCommunicationGroup(_defaultGroupName, _defaultGroupNumberOfTasks);
}
```

Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Network/Group/Driver/Impl/GroupCommDriver.cs]: Ah, never mind.

**Comment by markusweimer** [comment]
LGTM. Will test and merge.

**Comment by markusweimer** [comment]
tests pass, will merge.

**Pull request closed without merging**